targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/TARGET_FRDM/PeripheralNames.h

```diff
@@ -22,6 +22,14 @@
 extern "C" {
 #endif
 
+typedef enum {
+    GPIO_A = 0,
+    GPIO_B = 1,
+    GPIO_C = 2,
+    GPIO_D = 3,
+    GPIO_E = 4,
+} GPIOName;
```

Comment:
For peripherals like SPI and I2C you can only use one set of pins per peripheral. For GPIO you can use all the pins independently. Does specifying a peripheral gain us much? If not then this could be removed an a dummy peripheral value could be used for all pins.

Reply:
Done.